e2e/src/test/java/no/sysco/testing/kafka/e2e/ContainersIT.java

```diff
@@ -0,0 +1,194 @@
+    public static String JSON_SERVER_INSIDE_DOCKER_ENV;
+    public static String JSON_SERVER_EXPOSED;
+
+    public static GenericContainer httpProducer;
+    public static String HTTP_PRODUCER_EXPOSED;
+
+    public static GenericContainer httpMaterializer;
+
+    @BeforeClass
+    public static void start() {
+        setZookeeperAndKafka();
+        setSchemaRegistry();
+
+        setJsonServer();
+        setHttProducer();
+        setHttpMaterializer();
+    }
+
+    @Test
+    public void is_running() {
+        assertTrue(kafka.isRunning());
+        assertTrue(schemaRegistry.isRunning());
+
+        assertTrue(jsonServer.isRunning());
+        assertTrue(httpProducer.isRunning());
+        assertTrue(httpMaterializer.isRunning());
+
+        final List<MessageJsonRepresentation> messageJsonRepresentations =
+                Arrays.asList(
+                        RestAssured.given()
+                                .get(JSON_SERVER_EXPOSED + "/messages")
+                                .then()
+                                .statusCode(200)
+                                .extract()
+                                .as(MessageJsonRepresentation[].class));
+        assertTrue(messageJsonRepresentations.size() > 0);
+    }
+
+    @Test
+    public void test_data_pipeline_flow_successful() {
+        String id = UUID.randomUUID().toString();
+        String from = UUID.randomUUID().toString();
+        String to = UUID.randomUUID().toString();
+        String text = UUID.randomUUID().toString();
+
+        MessageJsonRepresentation messageJsonRepresentation =
+                new MessageJsonRepresentation(id, from, to, text);
+
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body(messageJsonRepresentation)
+                .post(HTTP_PRODUCER_EXPOSED + "/messages")
+                .then()
+                .statusCode(202);
+
+        await()
+                .atMost(70, TimeUnit.SECONDS)
+                .untilAsserted(
+                        () -> {
+                            MessageJsonRepresentation jsonPresentation =
+                                    RestAssured.given()
+                                            .get(JSON_SERVER_EXPOSED + "/messages/" + id)
+                                            .then()
+                                            // .statusCode(200)
+                                            .extract()
+                                            .as(MessageJsonRepresentation.class);
+
+                            assertNotNull(jsonPresentation);
+                        });
+    }
+
+    private static void setZookeeperAndKafka() {
+        kafka = new KafkaContainer(CONFLUENT_PLATFORM_VERSION);
+        kafka.start();
+    }
+
+    private static void setSchemaRegistry() {
+        // get network alias to be able connect other container to env
+        KAFKA_BROKER_INSIDE_DOCKER_ENV = "PLAINTEXT://" + kafka.getNetworkAliases().get(0) + ":9092";
```

Comment:
I think you can use the `kafka.getBootstrapServers()` method to get this.

Reply:
KafkaContainer use under the hood `SocatContainer proxy`, which makes `kafka.getBootstrapServers()` exposed to localhost, but inside docker network it uses `PLAINTEXT://<network-alias>:9092`. I have `KAFKA_BROKER_INSIDE_DOCKER_ENV`, which is bootstrap server inside docker network.